Api/csharp/ArmoniK.Api.Worker/Utils/WorkerServer.cs

```diff
@@ -0,0 +1,131 @@
+
+using Serilog;
+using Serilog.Formatting.Compact;
+
+namespace ArmoniK.Api.Worker.Utils;
+
+public class WorkerServer
+{
+  public static WebApplication Create<T>(IConfiguration? configuration = null)
+    where T : class
+  {
+    try
+    {
+      var builder = WebApplication.CreateBuilder();
+
+      builder.Configuration.SetBasePath(Directory.GetCurrentDirectory())
+             .AddJsonFile("appsettings.json",
+                          true,
+                          false)
+             .AddEnvironmentVariables();
+
+      if (configuration is not null)
+      {
+        builder.Configuration.AddConfiguration(configuration);
+      }
+
+      Log.Logger = new LoggerConfiguration().ReadFrom.Configuration(builder.Configuration)
+                                            .WriteTo.Console(new CompactJsonFormatter())
+                                            .Enrich.FromLogContext()
+                                            .CreateLogger();
+
+      var loggerFactory = LoggerFactory.Create(loggingBuilder => loggingBuilder.AddSerilog(Log.Logger));
+      var logger        = loggerFactory.CreateLogger("root");
+
+      builder.Host.UseSerilog(Log.Logger);
+
+
+      var computePlanOptions = builder.Configuration.GetRequiredSection(ComputePlan.SettingSection)
+                                      .Get<ComputePlan>();
+
+      if (computePlanOptions.WorkerChannel == null)
+      {
+        throw new Exception($"{nameof(computePlanOptions.WorkerChannel)} options should not be null");
+      }
+
+      builder.WebHost.ConfigureKestrel(options => options.ListenUnixSocket(computePlanOptions.WorkerChannel.Address,
+                                                                           listenOptions =>
+                                                                           {
+                                                                             if (File.Exists(computePlanOptions.WorkerChannel.Address))
+                                                                             {
+                                                                               File.Delete(computePlanOptions.WorkerChannel.Address);
+                                                                             }
+
+                                                                             listenOptions.Protocols = HttpProtocols.Http2;
+                                                                           }));
+
+      builder.Services.AddSingleton<ApplicationLifeTimeManager>()
+             .AddSingleton(_ => loggerFactory)
+             .AddSingleton<GrpcChannelProvider>()
+             .AddSingleton(computePlanOptions.AgentChannel)
+             .AddLogging()
+             .AddGrpc(options => options.MaxReceiveMessageSize = null);
+
+
+      var app = builder.Build();
+
+      if (app.Environment.IsDevelopment())
+      {
+        app.UseDeveloperExceptionPage();
+      }
+
+      app.UseSerilogRequestLogging();
+
+      app.UseRouting();
```

Comment:
Why ?

Reply:
The logging or the routing ? It seems we needed it at some point. I can retry without.

Reply:
Routing is necessary.